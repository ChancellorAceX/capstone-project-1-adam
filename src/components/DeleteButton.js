import React from 'react';
import '../styles/DeleteButton.css';
import TokenService from '../services/token-service';
import config from '../config';

export default class DeleteButton extends React.Component {

  deleteTarget = (e) => {
    const targetObject = { campaign: 'campaigns', encounter: 'encounters', character: 'encounterData' };
    e.preventDefault();
    fetch(`${config.API_ENDPOINT}/api/${this.props.target}/${this.props.id}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res => res.json())
      .catch(error => console.log({ error }));

    const targetState = targetObject[this.props.target];

    const filteredArray = this.props[targetState].filter(item => item.pcid !== this.props.id && item[0] !== this.props.id && item.eid !== this.props.id);
    
    this.props.setTargetState(targetState, filteredArray);
  };

  render() {
    return (
      <button className='deleteButton' type='button' onClick={this.deleteTarget}>
        &#128465;
      </button>
    );
  }
}