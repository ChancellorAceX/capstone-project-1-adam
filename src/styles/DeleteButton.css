button.deleteButton {
  width:12.5vw;
  height:5vh;
  font-size:22px;
  font-weight: 400;
  color: #b40a0a;
  background-color: #444444;
  display:flex;
  align-items: center;
  justify-content: center;
}