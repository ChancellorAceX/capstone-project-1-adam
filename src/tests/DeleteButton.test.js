import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeleteButton from '../components/DeleteButton';

describe('DeleteButton component', () => {
  const props = {
    "target": "character",
    "id": 16,
    "encounterData": [
      "{ac: 50, hp: 72, hpmax: 100, initiative: 30, level:…}",
      "{ac: 12, hp: 24, hpmax: 24, initiative: 5, level: n…}",
      "{ac: 12, hp: 15, hpmax: 20, initiative: null, level…}",
      "{ac: 18, hp: 32, hpmax: 32, initiative: null, level…}"
    ],
    "setTargetState": "ƒ () {}"
  };

  it('renders the full delete button component', () => {
    const wrapper = shallow(<DeleteButton {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});