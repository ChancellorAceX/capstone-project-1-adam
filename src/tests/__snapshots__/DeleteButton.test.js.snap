// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton component renders the full delete button component 1`] = `
<button
  className="deleteButton"
  onClick={[Function]}
  type="button"
>
  ðŸ—‘
</button>
`;
